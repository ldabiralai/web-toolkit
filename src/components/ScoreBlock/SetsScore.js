import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import * as colors from '../../colors';
import { fontFamilies } from '../../typography';
import * as breakpoints from '../../breakpoints';

const StyledTeamWrapper = styled.div`
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const StyledTeamName = styled.div`
  color: ${props => (props.hasWon === false ? colors.coreNeutral4 : colors.coreLightMinus1)};
  font-family: ${fontFamilies.alphaHeadline};
  font-size: 1em;
  flex-basis: 60%;
  text-transform: uppercase;
  span {
    font-family: ${fontFamilies.interUi};
    font-size: 1em;
  }
`;

export const StyledTeamSet = styled.span`
  color: ${props => (props.won ? colors.coreLightMinus1 : colors.coreNeutral4)};
  font-family: ${fontFamilies.alphaHeadline};
  font-weight: bold;
  font-size: 1em;
  margin: 0 0.4em;
  &:last-child {
    margin-right: 0px;
  }
  sup {
    font-size: 0.55em;
    vertical-align: super;
    position: relative;
    top: -0.4em;
    right: -0.2em;
  }
  ${breakpoints.small(css`
    margin: 0 0.8em;
  `)};
`;

const StyledTeamSets = styled.div`
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
`;

const Team = ({ teamData }) => {
  const { playerOneName, playerTwoName, sets, isServing, hasWon } = teamData;
  return (
    <StyledTeamWrapper>
      <StyledTeamName hasWon={hasWon}>
        <div>
          {playerOneName}
          {hasWon && <span> ✓</span>}
          {isServing && <span> •</span>}
        </div>
        <div>{playerTwoName}</div>
      </StyledTeamName>
      <StyledTeamSets>
        {sets &&
          sets.map(({ set, score, tie, ...props }) => (
            <StyledTeamSet key={`set-${set}`} {...props}>
              {score}
              {tie && <sup>{tie}</sup>}
            </StyledTeamSet>
          ))}
      </StyledTeamSets>
    </StyledTeamWrapper>
  );
};

const StyledWrapper = styled.div`
  width: 100%;
  ${props =>
    props.baseFontSize
      ? css`
          font-size: ${props.baseFontSize};
        `
      : css`
    font-size: 8px
    ${breakpoints.small(css`
      font-size: 14px;
    `)};
    ${breakpoints.medium(css`
      font-size: 24px;
    `)};
    ${breakpoints.large(css`
      font-size: 32px;
    `)};
    };
  `}
`;

export const StyledSpacer = styled.hr`
  display: block;
  height: 1px;
  border: 0;
  border-top: 1px solid;
  padding: 0;
  border-top-color: ${colors.coreLightMinus1};
`;

const SetsScore = ({ data, baseFontSize }) => (
  <StyledWrapper baseFontSize={baseFontSize} data-test="sets-score-wrapper">
    <Team teamData={data.topTeam} />
    <StyledSpacer />
    <Team teamData={data.bottomTeam} />
  </StyledWrapper>
);

const teamDataType = PropTypes.shape({
  hasWon: PropTypes.bool,
  isServing: PropTypes.bool,
  playerOneName: PropTypes.string.isRequired,
  playerTwoName: PropTypes.string,
  sets: PropTypes.arrayOf(
    PropTypes.shape({
      set: PropTypes.number,
      score: PropTypes.number,
      tie: PropTypes.number,
      won: PropTypes.bool,
    })
  ),
});

Team.propTypes = {
  teamData: teamDataType.isRequired,
};

SetsScore.defaultProps = {
  baseFontSize: null,
};

SetsScore.propTypes = {
  data: PropTypes.shape({
    bottomTeam: teamDataType,
    topTeam: teamDataType,
  }).isRequired,
  baseFontSize: PropTypes.string,
};

export default SetsScore;
