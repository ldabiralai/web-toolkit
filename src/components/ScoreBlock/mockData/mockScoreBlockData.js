export const pastMatchData = {
  topTeam: {
    playerOneName: 'Federer',
    sets: [
      { set: 1, score: 6, won: true, tie: null },
      { set: 2, score: 6, won: true, tie: null },
      { set: 3, score: 6, won: false, tie: 2 },
      { set: 4, score: 6, won: true, tie: null },
      { set: 5, score: 3, won: false, tie: null },
    ],
    hasWon: true,
    isServing: false,
  },
  bottomTeam: {
    playerOneName: 'Nadal',
    sets: [
      { set: 1, score: 3, won: false, tie: null },
      { set: 2, score: 3, won: true, tie: null },
      { set: 3, score: 7, won: true, tie: 7 },
      { set: 4, score: 3, won: false, tie: null },
      { set: 5, score: 6, won: true, tie: null },
    ],
    hasWon: false,
    isServing: false,
  },
};

export const liveMatchData = {
  topTeam: {
    playerOneName: 'MENENDEZ-MACEIRAS',
    playerTwoName: 'Rahman MAKSSOUD',
    sets: [
      { set: 1, score: 6, won: true, tie: null },
      { set: 2, score: 6, won: true, tie: null },
      { set: 3, score: 6, won: false, tie: 2 },
      { set: 4, score: 6, won: true, tie: null },
      { set: 5, score: 3, won: false, tie: null },
    ],
    hasWon: null,
    isServing: false,
  },
  bottomTeam: {
    playerOneName: 'MENENDEZ-MACEIRAS',
    playerTwoName: 'DAVIDOVICH FOKINA',
    sets: [
      { set: 1, score: 3, won: false, tie: null },
      { set: 2, score: 3, won: true, tie: null },
      { set: 3, score: 7, won: true, tie: 7 },
      { set: 4, score: 3, won: false, tie: null },
      { set: 5, score: 6, won: true, tie: null },
    ],
    hasWon: null,
    isServing: true,
  },
};

export const liveMatchDataSet = {
  topTeam: {
    playerOneName: 'A. MENENDEZ-MACEIRAS',
    playerTwoName: 'A. Rahman MAKSSOUD',
    sets: [
      { set: 1, score: 6, won: true, tie: null },
      { set: 2, score: 6, won: true, tie: null },
      { set: 3, score: 6, won: false, tie: 2 },
      { set: 4, score: 6, won: true, tie: null },
      { set: 5, score: 3, won: false, tie: null },
    ],
    hasWon: null,
    isServing: false,
  },
  bottomTeam: {
    playerOneName: 'A. MENENDEZ-MACEIRAS',
    playerTwoName: 'A. DAVIDOVICH FOKINA',
    sets: [
      { set: 1, score: 3, won: false, tie: null },
      { set: 2, score: 3, won: true, tie: null },
      { set: 3, score: 7, won: true, tie: 7 },
      { set: 4, score: 3, won: false, tie: null },
      { set: 5, score: 6, won: true, tie: null },
    ],
    hasWon: null,
    isServing: true,
  },
};
