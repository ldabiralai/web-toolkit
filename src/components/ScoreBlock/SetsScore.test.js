import React from 'react';
import { shallow, mount } from 'enzyme';
import SetsScore, { StyledTeamName, StyledTeamSet } from './SetsScore';
import { pastMatchData, liveMatchDataSet } from './mockData/mockScoreBlockData';

describe('<SetsScore />', () => {
  it('Renders a SetsScore component', () => {
    const wrapper = shallow(<SetsScore data={liveMatchDataSet} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with font size passed as prop', () => {
    const setScoreWrapper = shallow(<SetsScore data={liveMatchDataSet} baseFontSize="16px" />);
    expect(setScoreWrapper).toHaveStyleRule('font-size', '16px');
  });

  it('displays the winning team name with a checkmark', () => {
    const setsScore = mount(<SetsScore data={pastMatchData} />);
    const firstTeam = setsScore.find(StyledTeamName).first();
    expect(firstTeam.text().startsWith('Federer')).toBe(true);
    expect(firstTeam.text()).toContain(' ✓');
  });

  it('displays the second serving team with a dot', () => {
    const setsScore = mount(<SetsScore data={liveMatchDataSet} />);
    const secondTeam = setsScore.find(StyledTeamName).at(1);
    expect(secondTeam.text().startsWith('A. MENENDEZ-MACEIRAS')).toBe(true);
    expect(secondTeam.text()).toContain(' •');
  });

  it('renders the set scores', () => {
    const biggerWrapper = mount(<SetsScore data={liveMatchDataSet} />);
    expect(biggerWrapper.find(StyledTeamSet)).toHaveLength(10);
  });

  it('renders a tie score if any', () => {
    const biggerWrapper = mount(<SetsScore data={liveMatchDataSet} />);
    const thirdSet = biggerWrapper.find(StyledTeamSet).at(2);
    expect(thirdSet.exists('sup')).toEqual(true);
  });
});
