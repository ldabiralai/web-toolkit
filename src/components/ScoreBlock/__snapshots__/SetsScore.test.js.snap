// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetsScore /> Renders a SetsScore component 1`] = `
<Styled(div)
  baseFontSize={null}
  data-test="sets-score-wrapper"
>
  <Team
    teamData={
      Object {
        "hasWon": null,
        "isServing": false,
        "playerOneName": "A. MENENDEZ-MACEIRAS",
        "playerTwoName": "A. Rahman MAKSSOUD",
        "sets": Array [
          Object {
            "score": 6,
            "set": 1,
            "tie": null,
            "won": true,
          },
          Object {
            "score": 6,
            "set": 2,
            "tie": null,
            "won": true,
          },
          Object {
            "score": 6,
            "set": 3,
            "tie": 2,
            "won": false,
          },
          Object {
            "score": 6,
            "set": 4,
            "tie": null,
            "won": true,
          },
          Object {
            "score": 3,
            "set": 5,
            "tie": null,
            "won": false,
          },
        ],
      }
    }
  />
  <Styled(hr) />
  <Team
    teamData={
      Object {
        "hasWon": null,
        "isServing": true,
        "playerOneName": "A. MENENDEZ-MACEIRAS",
        "playerTwoName": "A. DAVIDOVICH FOKINA",
        "sets": Array [
          Object {
            "score": 3,
            "set": 1,
            "tie": null,
            "won": false,
          },
          Object {
            "score": 3,
            "set": 2,
            "tie": null,
            "won": true,
          },
          Object {
            "score": 7,
            "set": 3,
            "tie": 7,
            "won": true,
          },
          Object {
            "score": 3,
            "set": 4,
            "tie": null,
            "won": false,
          },
          Object {
            "score": 6,
            "set": 5,
            "tie": null,
            "won": true,
          },
        ],
      }
    }
  />
</Styled(div)>
`;
